source/postprocess/satellite_gravity.cc

```diff
@@ -0,0 +1,362 @@
+                  const double global_gy
+                    = Utilities::MPI::sum (local_gy, this->get_mpi_communicator());
+                  const double global_gz
+                    = Utilities::MPI::sum (local_gz, this->get_mpi_communicator());
+                  const double global_U
+                    = Utilities::MPI::sum (local_U, this->get_mpi_communicator());
+
+                  // calculate the gravity norm for each satellite position
+                  gnorm = std::sqrt((global_gx*global_gx)+  
+                                    (global_gy*global_gy)+  
+                                    (global_gz*global_gz));
+
+                  // analytical solution to estimate a mistfit according to satellite height 
+                  // can only be used if concentric density profile
+                  if (satellite_coordinate[0] <= model_inner_radius)
+                    {
+                      gtheory = 0;
+                      g_diff = -gnorm;
+                    }
+                  else if ((satellite_coordinate[0] > model_inner_radius) && (satellite_coordinate[0] < model_outer_radius))
+                    {
+                      gtheory = G * numbers::PI * 4/3 * reference_density * (satellite_coordinate[0] - (pow(model_inner_radius,3) / pow(satellite_coordinate[0],2)));
+                      g_diff = (gtheory - gnorm);
+                    }
+                  else
+                    {
+                      gtheory = G * numbers::PI * 4/3 * reference_density * (pow(model_outer_radius,3)-pow(model_inner_radius,3)) / pow(satellite_coordinate[0],2);
+                      g_diff = (gtheory - gnorm);
+                    }
+
+                  // write output
+                  f << satellite_coordinate[0] << ' '
+                    << satellite_coordinate[1] << ' '
+                    << satellite_coordinate[2] << ' '
+                    << position_satellite[0] << ' '
+                    << position_satellite[1] << ' '
+                    << position_satellite[2] << ' '
+                    << global_gx << ' '
+                    << global_gy << ' '
+                    << global_gz << ' '
+                    << gnorm << ' '
+                    << gtheory << ' '
+                    << g_diff << ' '
+                    << global_U
+                    << '\n';
+
+                }
+            }
+        }
+      return std::pair<std::string, std::string> ("gravity computation file:",
+                                                  filename);
+    }
+
+    template <int dim>
+    void
+    GravityCalculation<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Postprocess");
+      {
+        prm.enter_subsection ("Gravity calculation");
+        {
+          prm.declare_entry ("Mapping spherical spacing", "1",
+                             Patterns::Double ());
+          prm.declare_entry ("Profile depth spacing", "1",
+                             Patterns::Double ());
```

Comment:
If you put a number in like this: `Double(0.0)` it means the input parameter has to be larger than 0.0